alloc.h

```diff
@@ -5,7 +5,10 @@ struct alloc_state;
 struct tree;
 struct commit;
 struct tag;
-struct repository;
+struct repository
+{
+    int fetch_if_missing;
+}repo;
```

Comment:
In this header, this is just a forward declaration of `struct repository`. The actual struct is defined in `repository.h`. You probably want to add the field there.